### ISSUE TITLE
[BUG]: Cannot load module 'phalcon' because required module 'psr' is not loaded in Unknown on line 0

### ISSUE THREAD

**Issue by Essedum** [phalcon/cphalcon#14673]
Dears,
I have the following problem with installing phalcon on PHP 7.3 on Debian 10 + Apache.
Installed using this guide: https://docs.phalcon.io/4.0/en/installation#deb-based-distributions-debian-ubuntu-etc
With: sudo apt-get install php7.3-phalcon

I'm actually trying to install composer but it among other errors the first one it shows me the  is following:
_"Cannot load module 'phalcon' because required module 'psr' is not loaded in Unknown on line 0"_

The issue is that I have actually followed the [upgrade guide](https://docs.phalcon.io/4.0/en/upgrade#psr) and I do have psr installed. Actually phpinfo shows that both psr and phalcon are installed:
![image](https://user-images.githubusercontent.com/30523456/71646933-2c4cdb00-2cf7-11ea-8dc1-81c9a43f5c5f.png)
![image](https://user-images.githubusercontent.com/30523456/71646950-6a49ff00-2cf7-11ea-98c5-4628b0e1f5bc.png)

And I did put the psr extension prior to phalcon in php.ini file:
![image](https://user-images.githubusercontent.com/30523456/71646971-d7f62b00-2cf7-11ea-9899-afc00d1d9186.png)

Can you please help me out here. Because in order to fix the other issues I that the first thing you ask is php --ri phalcon output, but it gives the very same error:

_"PHP Warning:  Cannot load module 'phalcon' because required module 'psr' is not loaded in Unknown on line 0
Extension 'phalcon' not present"_

The output of php -v:
_"PHP Warning:  Cannot load module 'phalcon' because required module 'psr' is not loaded in Unknown on line 0
PHP 7.3.11-1~deb10u1 (cli) (built: Oct 26 2019 14:14:18) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.11, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.11-1~deb10u1, Copyright (c) 1999-2018, by Zend Technologies"_

So it looks like the system says that psr and phalcon are not installed and in the same time showing that they present. Tried to google more on this topic, but unfortunately, there are only 2 threads both leading nowhere as I have actually installed the psr.

Thank you very much in advance!
Appreciate your help.

**Comment by Jeckerson** [comment]
Did you tried to restart apache?

**Comment by ruudboon** [comment]
Have a look at the order. PSR must be loaded before Phalcon.
If you have multiple .ini files it helps to number them
```
10-psr.ini
20-phalcon.ini
```
if you have one extension.ini make sure the order is ok.
```
extension=psr.so
extension=phalcon.so
```

**Comment by Essedum** [comment]
> PHP CLI

Dear Jeckerson,

Thank you very much for this comment. It is solved now.
`php -m | egrep 'phalcon|psr' ` did not work as it gave the same error without anything else.
But I did `php -i |grep php\.ini` to find where is the php.ini for PHP CLI and it appeared to be the default one. I copy-pasted the lines from my php.ini that was in the original post. Restarted apache and everything worked.

Thank you so much!

**Comment by Chealer** [comment]
For those using a packagecloud package, the issue is under discussion in [this forum thread](https://forum.phalcon.io/discussion/19779/packagecloudio-repositories-is-pointless).

**Comment by Chealer** [comment]
The discussion I mentioned above is over (the forum is now read-only).

For Ubuntu, this can be solved by upgrading to version 20.04 or ulterior and installing the php-psr package. Otherwise, a simple workaround is to downgrade to Phalcon 3. For example, in Ubuntu 18.04:
`# apt install php7.2-phalcon=3.4.5-1+php7.2`

**Comment by ekolga** [comment]
> The discussion I mentioned above is over (the forum is now read-only).
> 
> For Ubuntu, this can be solved by upgrading to version 20.04 or ulterior and installing the php-psr package. Otherwise, a simple workaround is to downgrade to Phalcon 3. For example, in Ubuntu 18.04: `# apt install php7.2-phalcon=3.4.5-1+php7.2`

Thank you so much for your comment. Really helped me out
